test/src/test/java/org/corfudb/runtime/collections/PersistentCorfuTableTest.java

```diff
@@ -432,20 +437,31 @@ public void testUncommittedChangesIsolationBetweenParallelTxns() throws Exceptio
             } catch (InterruptedException e) {
                 e.printStackTrace();
             }
+
             // Changes made by writeThread should be isolated.
             readerResult.set(corfuTable.get(key1).getPayload().getLsb());
+
+            // Read is done. Signal the writerThread to commit
+            writeLatch.countDown();
             rt.getObjectsView().TXEnd();
         });
 
         // put(k1, v2) to overwrite the previous put, but do not commit
         Thread writerThread = new Thread(() -> {
             rt.getObjectsView().TXBegin();
-            CorfuRecord value2 = new CorfuRecord(payload2, metadata1);
+            CorfuRecord<TestSchema.Uuid, TestSchema.Uuid> value2 = new CorfuRecord<>(payload2, metadata1);
             corfuTable.insert(key1, value2);
             writerResult.set(corfuTable.get(key1).getPayload().getLsb());
 
             // Signals the readerThread to read
             readLatch.countDown();
+
+            try {
+                // Unblocked until readThread starts txn
```

Comment:
I would just update this comment to make it more clear that we are unblocked only after the read is performed.